spec/mail/securityconsiderations.mdown

```diff
@@ -10,7 +10,32 @@ Automatic conversion of charsets that allow hidden channels for ASCII text, such
 
 To allow the client to restrict the volume of data it can receive in response to a request, a maximum length may be requested for the data returned for a textual body part. However, truncating the data may change the semantic meaning, for example truncating a URL changes its location. Servers that scan for links to malicious sites should take care to either ensure truncation is not at a semantically significant point, or to rescan the truncated value for malicious content before returning it.
 
-## Email Submission
+## HTML email display
+
+HTML message bodies provide richer formatting for emails but present a number of security challenges especially when embedded in a webmail context in combination with interface HTML. Clients that render HTML email should make careful consideration of the potential risks, including:
+
+* If JavaScript is allowed to execute it can rewrite email to change its
+  content on subsequent opening, allowing users to be mislead. In webmail systems, if run in the same origin as the interface it can access and exfiltrate all user content accessible via the browser, including all other emails and potentially contacts, calendar events, settings, and credentials. It can also rewrite the interface to undetectably phish passwords. A compromise is likely to be persistent, not just for the duration of page load, due to exfiltration of session credentials or installation of a service worker that can intercept all subsequent network requests (this however would only be possible if blob downloads are also available on the same origin, and the service worker script is attached to the message).
+* HTML documents may load content directly from the internet, rather than just
+  referencing attached resources. For example you may have an `<img>` tag with an external `src` attribute. This may leak to the sender when a message is opened, as well as the IP address of the recipient. Cookies may also be sent and set by the server, allowing tracking between different emails and even website visits and advertising profiles.
+* In webmail systems, CSS can break the layout or create phishing
+  vulnerabilities. For example, the use of `position:fixed` can allow an email to draw content outside of its normal bounds, potentially clickjacking a real interface element.
+* If not inside a separate frame in a webmail context, any styles defined in
+  CSS rules will apply to interface elements as well if the selector matches, allowing the interace to be modified. Similarly, any interface styles that match elements in the email will alter their appearance, potentially breaking the layout of the email.
+* The link text in HTML has no neccessary correlation with the actual target of
+  the link, which can be used to make phishing attacks more convincing.
```

Comment:
links should be rewritten with `target=_new` and [`rel=noopener`](https://mathiasbynens.github.io/rel-noopener/)

Reply:
Is there an opportunity here to highlight the potential risks of punycode internationalized domain names (e.g. `xn--80ak6aa92e.com` renders as `apple.com`) and hyperlinks to them? While topically relevant to these list items in the security considerations document, I'm not sure how much can be done to mitigate them within the JMAP spec or if it's even appropriate to try to do so. If nothing else, I think it's a positive thing to continue to remind stakeholders of the risks posed by punycode IDNs.

Reply:
Personally... I just don't see that problem. The phishing mail I get comes from laughable things like support@apple-security-team.cx and applesecurityreview2018@ymail.com, which suggests to me that fools can be separated from their money without any unicode shenanigans.

Reply:
While I'd like us to be in a world where people pay attention to the sender, my experience says that phishers get away with everything anyway.